src/nvim/option.c

```diff
@@ -4667,7 +4664,9 @@ char_u *check_colorcolumn(win_T *wp)
       col = col * getdigits_int(&s);
       if (wp->w_buffer->b_p_tw == 0)
         goto skip;          /* 'textwidth' not set, skip this item */
-      col += wp->w_buffer->b_p_tw;
+      assert(wp->w_buffer->b_p_tw >= (long)INT_MIN - col
+             && wp->w_buffer->b_p_tw <= (long)INT_MAX - col);
```

Comment:
Added review commit with some fixed asserts in option.c.
I think there are other fixable asserts in other files, but can't continue working just now. I'll resume later this evening.